scipy/special/_support_alternative_backends.py

```diff
@@ -82,11 +79,13 @@ def __xlogy(x, y, *, xp=xp):
 
 
 def _chdtr(xp, spx):
+    if spx is None:
```

Comment:
If `scipylike_namespace_for` returns `None` as the default, we get an error if we forget to include this logic. If `scipylike_namespace_for` returns `scipy` as the default, then we end up using `scipy.special.gammainc` to define `scipy.special.chdtr` instead of just falling back to `scipy.special.chdtr`. I think this was happening with `array_api_strict` before.